File: Part 5 - Association Rule Learning/Section 28 - Apriori/apriori_me.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
df=pd.read_csv('Market_Basket_Optimisation.csv',header=None)
transactions=[]
for i in range(0,7501):
    transactions.append([str(df.values[i,j]) for j in range(0,20)])

# Training Apriori on the dataset
from apyori import apriori
rules = apriori(transactions,min_support=0.003,min_confidence=0.2,min_lift=3,min_length = 2)

# Visualising the dataset
results=list(rules)

def inspect(results):
    rh=[tuple(result[2][0][0]) for result in results]
    lh=[tuple(result[2][0][1]) for result in results]
    supports=[result[1] for result in results]
    confidences=[result[2][0][2] for result in results]
    lifts=[result[2][0][3] for result in results]
    return list(zip(rh,lh,supports,confidences,lifts))

resultDataFrame=pd.DataFrame(inspect(results))

File: Part 5 - Association Rule Learning/Section 28 - Apriori/apriori_me.R
dataset=read.csv('Market_Basket_Optimisation.csv',header = FALSE)
# install.packages('arules')
library(arules)
dataset=read.transactions('Market_Basket_Optimisation.csv',sep = ',',rm.duplicates = TRUE)
#Sparsing matrix the data

#info visualising and details
summary(dataset)
itemFrequencyPlot(dataset,topN=10)

# Training Apriori on the dataset
rules = apriori(data = dataset,parameter = list(support = 0.004,confidence = 0.2))
# atleast purchased 3 times a day 3*7/7500 minimum support, default confidence 0.8.
# then keep decreasing to smaller confidence.

#Visualising the results
inspect(sort(rules,by = 'lift')[1:10])

